### PR TITLE
Paragraphs support

### PR DESCRIPTION
**Pull request by damienmckenna**
From alexej-d.

### CODE REVIEW

File: multiversion.module [hunk 1]
```diff
@@ -67,6 +67,11 @@ function multiversion_entity_type_alter(array &$entity_types) {
           $entity_type->setHandlerClass('storage', "$namespace\\MediaStorage");
           $entity_type->setHandlerClass('storage_schema', "$namespace\\ContentEntityStorageSchema");
           break;
+
```
Comment (damienmckenna) [inline multiversion.module]: There's unneeded whitespace on this line.

File: src/Entity/Storage/Sql/ParagraphsStorage.php [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  public function save(EntityInterface $entity) {
+    $result = $this->saveEntity($entity);
+    $this->updateRevIds($entity);
+
+    return $result;
+  }
+
+  // change revision id the parent entity links to
```
Comment (damienmckenna) [inline src/Entity/Storage/Sql/ParagraphsStorage.php]: The comments need to be updated to match the Drupal coding standards.

File: src/Entity/Storage/Sql/ParagraphsStorage.php [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    if($parent_entity = $entity->getParentEntity()){
+      $parent_entity_id = $parent_entity->id();
+      $parent_entity_rev_id = $parent_entity->getRevisionId();
+      $parent_entity_type = $parent_entity->getEntityTypeId();
+      $parent_entity_field_name = $entity->parent_field_name->value;
+
+      $entity_id = $entity->id();
+      $entity_rev_id = $entity->getRevisionId();
+
+      if(!$parent_entity_id   ||
```
Comment (damienmckenna) [inline src/Entity/Storage/Sql/ParagraphsStorage.php]: The if() statements need to be tidied up to match Drupal's coding standards.

**Comment by damienmckenna** [comment]
I uploaded the suggested improvements to a patch to https://www.drupal.org/node/2756417.

**Comment by jeqq** [comment]
This PR is outdated, Multiversion already supports Paragraphs.

**Pull request closed without merging**